app/uk/gov/hmrc/helptosavefrontend/health/NSIConnectionHealthCheck.scala

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosavefrontend.health
+
+import java.time.LocalDate
+
+import akka.actor.{Actor, ActorRef, ActorSystem, Props}
+import com.google.inject.Inject
+import configs.syntax._
+import play.api.Configuration
+import uk.gov.hmrc.helptosavefrontend.connectors.NSIConnector
+import uk.gov.hmrc.helptosavefrontend.health.NSIConnectionHealthCheck.NSIConnectionHealthCheckRunner
+import uk.gov.hmrc.helptosavefrontend.health.NSIConnectionHealthCheck.NSIConnectionHealthCheckRunner.Payload
+import uk.gov.hmrc.helptosavefrontend.metrics.Metrics
+import uk.gov.hmrc.helptosavefrontend.models.NSIUserInfo
+import uk.gov.hmrc.helptosavefrontend.models.NSIUserInfo.ContactDetails
+import uk.gov.hmrc.helptosavefrontend.util.{Email, Logging}
+import uk.gov.hmrc.http.HeaderCarrier
+
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+
+class NSIConnectionHealthCheck @Inject() (system: ActorSystem, configuration: Configuration, metrics: Metrics, nSIConnector: NSIConnector) extends Logging {
+
+  val name: String = "nsi-connection"
+
+  val enabled: Boolean = configuration.underlying.get[Boolean](s"health.$name.enabled").value
+
+  lazy val healthCheck: ActorRef = system.actorOf(
+    HealthCheck.props(
+      name,
+      configuration.underlying,
+      system.scheduler,
+      metrics.metrics,
+      () ⇒ (),
+      () ⇒ (),
+      NSIConnectionHealthCheckRunner.props(nSIConnector, metrics, Payload.Payload1),
+      NSIConnectionHealthCheckRunner.props(nSIConnector, metrics, Payload.Payload2)
+    )
+  )
+
+  // start the health check only if it is enabled
+  if (enabled) {
+    logger.info(s"$name health test enabled - starting HealthCheck actor")
```

Comment:
How about...
"HealthCheck for $name is enabled, commencing connection checking period timer every XXXs"
...plus can we change the word 'test' or 'Test' in the code to check